subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportImplementation.kt

```diff
@@ -196,49 +195,55 @@ class ReportImplementation(
     @Synchronized
     override fun reportTestCase(): ReportState.Initialized.Started =
         methodExecutionTracing("reportTestCase") {
-            val currentState = state
-            if (currentState !is ReportState.Initialized.Started) {
+            val startedState = state
+            if (startedState !is ReportState.Initialized.Started) {
                 throw RuntimeException("Reporter has invalid state. Expected state: Started. Actual state: $state")
             }
+            startedState.endTime = timeProvider.nowInSeconds()
 
-            currentState.apply {
-                currentState.endTime = timeProvider.nowInSeconds()
-                afterTestStop(this)
+            try {
+                afterTestStop(startedState)
```

Comment:
Because of error in a test `Stop` callback test became unwritten
I prefer to log that Throwable to FAIL whole test